pennylane/hf/tapering.py

```diff
@@ -0,0 +1,204 @@
+        currcol[irow] = 0
+        binary_matrix[:, icol:] ^= np.outer(currcol, pvtcols)
+
+    return binary_matrix.astype(int)
+
+
+def _kernel(binary_matrix):
+    """ Computes the kernel of a binary matrix on the binary finite field Z_2 """
+
+    # Get the columns with and without pivots
+    pivots = (binary_matrix.T!=0).argmax(axis=0)
+    nonpivots = np.setdiff1d(range(len(binary_matrix[0])), pivots)
+
+    # Initialize the nullspace
+    null_vector = np.zeros((binary_matrix.shape[1], len(nonpivots)), dtype=int)
+    null_vector[nonpivots, np.arange(len(nonpivots))] = 1
+
+    # Fill up the nullspace vectors from the binary matrix
+    null_vector_indices = np.ix_(pivots, np.arange(len(nonpivots)))
+    binary_vector_indices = np.ix_(np.arange(len(pivots)), nonpivots)
+    null_vector[null_vector_indices] = -binary_matrix[binary_vector_indices]%2
+
+    return null_vector.T
+
+
+def generate_taus(nullspace, num_qubits):
+    """Generate generators tau from the nullspace
+
+    Args:
+        nullspace (list): kernel of the binary matrix corresponding to the Hamiltonian.
+        num_qubits (int): number of wires required to define the Hamiltonian.
+
+    Returns:
+        generators (list): list of generators of symmetries, taus, for the Hamiltonian.
+
+    """
+
+    generators = []
+
+    for null_vector in nullspace:
+        tau = qml.Identity(0)
+        for idx, op in enumerate(
+            zip(null_vector[:num_qubits], null_vector[num_qubits:])
+        ):
+            x, z = op
+            if x == 0 and z == 0:
+                tau @= qml.Identity(idx)
+            elif x == 1 and z == 0:
+                tau @= qml.PauliX(idx)
+            elif x == 1 and z == 1:
+                tau @= qml.PauliY(idx)
+            else:
+                tau @= qml.PauliZ(idx)
+        ham = qml.Hamiltonian([1.0], [tau], simplify=True)
+        generators.append(ham)
+    return generators
+
+
+def generate_paulis(generators, num_qubits):
+    """Generate the single qubit Pauli X operators :math:`sigma^{x}_{i}` that will be used for
+    for generating cliffords for a Hamiltonian :math:`H`.
+
+    Args:
+        generators (list): list of generators of symmetries, taus, for the Hamiltonian.
+        num_qubits (int): number of wires required to define the Hamiltonian.
+    Return:
+        sigma_x (list): the list of support of the single-qubit PauliX operators used to build
+                        the Clifford operators
+
+    .. code-block::
+        >>> symbols, coordinates = (['H', 'H'], np.array([0., 0., -0.66140414, 0., 0., 0.66140414]))
+        >>> H, qubits = qml.qchem.molecular_hamiltonian(symbols, coordinates)
+        >>> generators = generate_symmetries(H, qubits)
+        >>> generate_clifford(generators, qubits)
+        [PauliX(wires=[1]), PauliX(wires=[2]), PauliX(wires=[3])]
+
+    """
+    ops_generator = [g.ops[0] if isinstance(g.ops, list) else g.ops for g in generators]
```

Comment:
This is an example of why we could benefit from a more consistent format for observables that doesn't change whetehr we have one or many terms, right?

Reply:
Yes.